tests/test_proxy_functional.py

```diff
@@ -1,14 +1,137 @@
 import asyncio
 import os
 import pathlib
+import ssl
 from unittest import mock
 
+import proxy
 import pytest
 from yarl import URL
 
 import aiohttp
 from aiohttp import web
 
+# @pytest.fixture
+# def secure_proxy_url(
+#     aiohttp_unused_port, tls_ca_certificate_pem_path, tls_ca_private_key_pem_path
+# ):
+#     """Return the an instance of a running secure proxy."""
+#     unused_port = str(aiohttp_unused_port())
+#     address = "127.0.0.1"
+#     input_args = [
+#         "--num-workers",
+#         "1",
+#         "--hostname",
+#         "127.0.0.1",
+#         "--port",
+#         unused_port,
+#         "--cert-file",
+#         tls_ca_certificate_pem_path,
+#         "--key-file",
+#         tls_ca_private_key_pem_path,
+#     ]
+#     proxy_url = URL.build(scheme="https", host=address, port=unused_port)
+#     with proxy.Proxy(input_args=input_args):
+#         yield proxy_url
+
+
+@pytest.fixture
+def secure_proxy_url(
+    aiohttp_unused_port,
+    tls_certificate,
+):
+    """Return the an instance of a running secure proxy."""
+    unused_port = str(aiohttp_unused_port())
+    address = "127.0.0.1"
+    with tls_certificate.cert_chain_pems[0].tempfile() as server_cert_path:
+        with tls_certificate.private_key_pem.tempfile() as server_key:
+            input_args = [
+                "--num-workers",
+                "1",
+                "--hostname",
+                "127.0.0.1",
+                "--port",
+                unused_port,
+                "--cert-file",
+                server_cert_path,
+                "--key-file",
+                server_key,
+            ]
+            proxy_url = URL.build(scheme="https", host=address, port=unused_port)
+            with proxy.Proxy(input_args=input_args):
+                yield proxy_url
+
+
+# @pytest.mark.xfail
+async def test_secure_proxy_http_absolute_path(
+    loop,
+    aiohttp_server,
+    client_ssl_ctx,
+    secure_proxy_url,
+) -> None:
+    """Test urls can be requested through a secure proxy."""
+    response_body = "Test message"
+
+    async def handler(*args, **kwargs):
+        return web.Response(text=response_body)
+
+    app = web.Application()
+    app.router.add_route("GET", "/", handler)
+    server = await aiohttp_server(app)
+
+    conn = aiohttp.TCPConnector(loop=loop)
```

Comment:
I tried finding another fixture to use instead of instantiating the `TCPConnector` and `ClientSession` here, but those all configured the SSL context more than I needed when making the `get()` so I ended up writing this.